File: recursion/SkipCharacter.java
package recursion;

public class SkipCharacter {
    public static void main(String[] args) {
        System.out.println(skipCharacter("baccad"));
    }

    static String skipCharacter(String str) {
        return helper(str).toString();
    }

    static StringBuilder helper(String str) {
        StringBuilder sb = new StringBuilder();

        if(str.charAt(0) != 'a') sb.append(str.charAt(0));

        if (str.length() == 1) {
            return sb;
        }

        return sb.append(helper(str.substring(1)));
    }
}

File: recursion/SkipCharacter.py
def skip_character(string, skip):
    return helper(string, skip)


def helper(string, skip):
    result = ''

    if string[0] != 'a':
        result += string[0]

    if len(string) == 1:
        return result

    return result + helper(string[1:], skip)


print(skip_character('baccad', 'a'))
